idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/markers/predef.kt

```diff
@@ -0,0 +1,61 @@
+package arrow.meta.ide.plugins.proofs.markers
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.internal.Noop
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.codeInsight.daemon.MergeableLineMarkerInfo
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.renderer.DescriptorRenderer
+import java.awt.event.MouseEvent
+import javax.swing.Icon
+
+/**
+ * Similar to [arrow.meta.ide.dsl.editor.lineMarker.LineMarkerSyntax.addLineMarkerProvider], is an extension for
+ * PsiElements that are not leafs so it will look for the first Leaf corresponding the targeted psiElement
+ */
+@Suppress("UNCHECKED_CAST")
+fun <A : PsiElement> IdeMetaPlugin.addLineMarkerProvider(
```

Comment:
```suggestion
internal fun <A : PsiElement> IdeMetaPlugin.addLineMarkerProvider(
```

Reply:
👍

Reply:
Excuse that. 
We'll open a refactoring task to unify all those variants.